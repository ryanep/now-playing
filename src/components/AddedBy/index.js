import React from "react";
import styles from "./style.scss";

export default ({ name, photo }) => (
  <div className={styles.info}>
    <div className={styles.added}>
      <div className={styles.addedName}>Added by</div>
      <div id="added-by-name" className={styles.name}>{name}</div>
    </div>
    {photo && <img className={styles.photo} src={photo} alt={name} />}
  </div>
);
