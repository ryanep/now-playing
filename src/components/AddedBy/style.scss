.info {
	min-width: 20rem;
	text-align: right;
}

.added {
    display: inline-block;
    vertical-align: middle;
    text-shadow: 0 2px 2px rgba(0, 0, 0, 0.2);
}

.addedName {
    font-size: 1.4rem;
}

.name {
    font-weight: bold;
}

.photo {
    display: inline-block;
    vertical-align: middle;
    margin-left: 1rem;
    width: 3.5rem;
    height: auto;
    border-radius: 0.4rem;
    box-shadow: 0 1px 1px rgba(0, 0, 0, 0.2);
}