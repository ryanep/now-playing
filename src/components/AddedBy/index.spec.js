import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import AddedBy from "./index";

describe("components/AddedBy", () => {
  const name = "__name__";
  const photo = "__photo__";

  // Snapshot testing
  it("renders the name and photo provided", () => {
    const component = renderer.create(<AddedBy name={name} photo={photo} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders only the name when no photo is provided", () => {
    const component = renderer.create(<AddedBy name={name} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // DOM Assertion
  it("renders the name provided", () => {
    const component = shallow(<AddedBy name={name} />);
    expect(component.find("#added-by-name").text()).toEqual("__name__");
  });

  it("renders only the name when only a name is provided", () => {
    const component = shallow(<AddedBy name={name} />);
    expect(component.find("img")).toHaveLength(0);
  });

  it("renders the name and photo provided", () => {
    const component = shallow(<AddedBy name={name} photo={photo} />);
    expect(component.find("#added-by-name").text()).toEqual("__name__");
    expect(component.find("img").prop("src")).toEqual("__photo__");
    expect(component.find("img").prop("alt")).toEqual("__name__");
  });
});
