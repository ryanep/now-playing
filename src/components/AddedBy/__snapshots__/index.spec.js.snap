// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/AddedBy renders only the name when no photo is provided 1`] = `
<div
  className="info"
>
  <div
    className="added"
  >
    <div
      className="addedName"
    >
      Added by
    </div>
    <div
      className="name"
      id="added-by-name"
    >
      __name__
    </div>
  </div>
</div>
`;

exports[`components/AddedBy renders the name and photo provided 1`] = `
<div
  className="info"
>
  <div
    className="added"
  >
    <div
      className="addedName"
    >
      Added by
    </div>
    <div
      className="name"
      id="added-by-name"
    >
      __name__
    </div>
  </div>
  <img
    alt="__name__"
    className="photo"
    src="__photo__"
  />
</div>
`;
